docs/hokusai_files.md

```diff
@@ -56,19 +56,21 @@ Please see [hokusai configure command](./Command_Reference.md#configuring-hokusa
 - `post-deploy`: <string> (optional) - A post-deploy hook - useful for deploy notifications
-- `run-tty`: <boolean> (optional) - Attach the terminal to your shell session when invoking `hokusai [staging|production|review_app] run`.  Bound to the `--tty` option for this command, and falls back to by the `HOKUSAI_RUN_TTY` env var.
-    - `follow-logs`: <boolean> (optional) - Follow log output when invoking `hokusai [staging|production|review_app] logs`.  Bound to the `--follow` option for this command, and falls back to the `HOKUSAI_FOLLOW_LOGS` env var.
-    - `tail-logs`: <integer> (optional) - Tail N lines of log output when invoking `hokusai [staging|production|review_app] logs`.  Bound to the `--tail` option for this command, and falls back to the `HOKUSAI_TAIL_LOGS` env var.
+- `pre-build`: <string> (optional) - A pre-build hook - useful to inject dynamic environment variables into the build, for example: `export COMMIT_HASH=$(git rev-parse HEAD)`
+- `pre-deploy`: <string> (optional) - A pre-deploy hook - useful to enforce migrations
+- `project-name`: <string> (required) - The project name
 - `run-constraints`: <list of kubernetes label selector strings> - Constrain run containers to Kubernetes nodes matching the label selectors in the form `key=value` by setting the `nodeSelector` field on the container's spec. Bound to the `--constrint` option for `hokusai [staging|production|review_app] run` as well as containers run via the `--migration` flag as well as `pre-deploy` / `post-deploy` hooks triggered by `hokusai [staging|production|review_app] deploy` or `hokusai pipeline promote`.  Falls back to the `HOKUSAI_RUN_CONSTRAINTS` env var, in which case a list is parsed from a comma-delimited string.
-- `always-verbose`: <boolean> (optional) - Always print verbose output.  Bound to the `--verbose` option for various commands, and falls back to the `HOKUSAI_ALWAYS_VERBOSE` env var.
+- `run-template`: <string> (optional) - `hokusai [staging|production|review_app] run` creates a pod to run the user-specified command. This config specifies name of the Deployment in `hokusai/staging.yml` or `hokusai/production.yml` to model the run-pod's spec after. If not specified, Hokusai uses the first Deployment found in those files.
+- `run-tty`: <boolean> (optional) - Attach the terminal to your shell session when invoking `hokusai [staging|production|review_app] run`.  Bound to the `--tty` option for this command, and falls back to by the `HOKUSAI_RUN_TTY` env var.
+- `secrets-file`: <string> (optional): A project may rely on secret app configs that are loaded from a file into the shell environment of its app container. This config specifies the path to that file. The content of the file must be in `export KEY='VALUE'` form. `hokusai [staging|production|review_app] run` reads this config and takes care of `source`ing the file, in addition to running the user-specified command.
```

Comment:
new

Reply:
Oh yes, secrets file path will likely be the same from project to project, so does seems it should be a global config. If so, we want a way to migrate projects gracefully, so that `run` still works with projects not yet using a secrets file.

Reply:
removed `/secrets/secrets` related code. will follow up on future PR.